src/Microsoft.OData.Client/Serialization/PrimitiveXmlConverter.cs

```diff
@@ -858,10 +858,36 @@ internal override string ToString(object instance)
         }
     }
 
+    /// <summary>
+    /// Convert between primitive types Edm.Date and string
+    /// </summary>
+    internal sealed class DateOnlyTypeConverter : PrimitiveTypeConverter

```

Comment:
Thanks for pointing out this. Since they are internal and small enough and Please Let me keep them as now. It's same for others in the same file. We'd like to do it in the next fully refactor version.